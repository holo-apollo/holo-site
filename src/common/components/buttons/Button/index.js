import React from 'react';
import styled, { css } from 'styled-components';

import palette from 'common/palette';

const getButtonProps = ({ width, height, color, fontSize, fontWeight }) => css`
  width: ${width}px;
  min-width: ${width}px;
  height: ${height}px;
  line-height: ${height}px;
  background-color: ${color};
  font-size: ${fontSize}px;
  font-weight: ${fontWeight};
`;

const getHoverProps = ({ hoverColor }) => css`
  display: inline-block;
  vertical-align: middle;
  transform: perspective(1px) translateZ(0);
  box-shadow: 0 0 1px transparent;
  position: relative;
  transition-property: background-color;
  transition-duration: 0.5s;

  &:before {
    content: '';
    position: absolute;
    z-index: -1;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: ${hoverColor};
    transform: scaleX(0);
    transform-origin: 100% 50%;
    transition-property: transform;
    transition-duration: 0.5s;
    transition-timing-function: ease-out;
  }

  &:hover,
  &:focus,
  &:active {
    color: white;
    border: none;

    &:before {
      transform: scaleX(1);
      transition-timing-function: cubic-bezier(0.52, 1.64, 0.37, 0.66);
    }
  }
`;

const StyledButton = styled.button`
  font-weight: 500;
  text-align: center;
  color: ${palette.white};
  cursor: pointer;
  padding: 0;
  margin: 0;
  border: none;
  border-radius: 0;
  -webkit-appearance: none;

  &:focus {
    outline: none;
  }

  ${getButtonProps};
  ${getHoverProps};
`;

const Button = props => <StyledButton {...props} />;

Button.defaultProps = {
  color: palette.darkGrey,
  hoverColor: palette.blue,
  height: 35,
  width: 250,
  fontSize: 14,
  fontWeight: '500',
};

export default Button;
