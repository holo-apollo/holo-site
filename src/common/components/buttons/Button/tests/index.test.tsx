import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../index';

describe('Button', function() {
  it('should match snapshot', function() {
    const tree = renderer.create(<Button>Button text</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
