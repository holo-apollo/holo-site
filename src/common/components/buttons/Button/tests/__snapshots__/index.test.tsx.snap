// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<button
  className="Button__StyledButton-sc-19ftob3-0 ezLdBm"
  color="#161616"
  fontSize={18}
  fontWeight="bold"
  height={35}
  width={250}
>
  Button text
</button>
`;
